app/views/admin/lessons/show.html.erb

```diff
@@ -44,5 +44,7 @@
   <%= link_to 'Download', admin_download_lesson_path(@lesson), disable_with: 'Downloading...' %>
 </p>
 
+<% if (@lesson.user == current_user) || current_user.admin? || current_user.super_user? %>
```

Comment:
nice! 

We could create a helper method  (or even a Pundit policy) to encapsulate the authorization logic, as it looks like those authorization checks are used in multiple places. However, this is perfectly fine for the MVP.